Service/ComponentenCatalogusService.php

```diff
@@ -35,9 +55,227 @@ public function __construct(
+     *
+     * @return array
+     */
+    public function getApplication(string $id){
+
+        // Dow e have a source
+        if(!$source = $this->getSource()){
+            return;
+        }
+
+        $this->io->success('Getting application '.$id);
+        $response = $this->callService->call($source, '/products/'.$id);
+
+        $application = json_decode($response->getBody()->getContents(), true);
+
+        if(!$application){
+            $this->io->error('Could not find an application with id: '.$id.' and with source: '.$source->getName());
+            return;
+        }
+        $application = $this->importApplication($application);
+
+        $this->entityManager->flush();
+
+        $this->io->success('Found application with id: '.$id);
+
+        return $application->toArray();
+    }
+
+    /**
+     * @return ObjectEntity
+     */
+    public function importApplication($application){
+
+        // Dow e have a source
+        if(!$source = $this->getSource()){
+            return ;
+        }
+        if(!$applicationEntity = $this->getApplicationEntity()){
+            return ;
+        }
+
+        $this->io->success("Checking application ".$application['name']);
+        $synchronization = $this->synchronizationService->findSyncBySource($source, $applicationEntity, $application['id']);
+        $synchronization = $this->synchronizationService->handleSync($synchronization, $application);
+
+        return $synchronization->getObject();
+    }
+
+    /**
+     * Get the component entity
+     *
+     * @return ?Entity
+     */
+    public function getComponentEntity(): ?Entity{
+
+        if(!$this->componentEntity = $this->entityManager->getRepository("App:Entity")->findOneBy(["reference"=>"https://opencatalogi.nl/oc.component.schema.json"])){
+            $this->io->error("No entity found for https://opencatalogi.nl/oc.component.schema.json");
+        }
+
+        return $this->componentEntity;
+    }
+
+    /**
+     * Get the component mapping
+     *
+     * @return ?Mapping
+     */
+    public function getComponentMapping(): ?Mapping{
+        if(!$this->componentMapping = $this->entityManager->getRepository("App:Mapping")->findOneBy(["reference"=>"https://componentencatalogus.commonground.nl/api/components"])){
+            $this->io->error("No mapping found for https://componentencatalogus.commonground.nl/api/components");
+        }
+
+        return $this->componentMapping;
+    }
+
+
+    /**
+     * Get components through the components of https://componentencatalogus.commonground.nl/api/components
+     *
+     * @return array
+     */
+    public function getComponents(): array{
+
+        $result = [];
+        // Dow e have a source
+        if(!$source = $this->getSource()){
+            return $result;
+        }
+
+        $this->io->comment('Trying to get all components from source '.$source->getName());
+
+        // @TODO rows per page are 10, so i get only 10 results
```

Comment:
Maybe we should get multiple pages here?
Example of this in the SynchronizationService